### PR TITLE
[13.0][IMP] account_due_list: Better UX in view, filters and action context

### PR DESCRIPTION
**Pull request by carlosdauden**
- Refactor tree view to add fields, use optional show and improve decorators.
 - Refactor and improve filters.
 - Add default search with new filter to avoid that unposted account move lines are showed.

@Tecnativa TT28007

### CODE REVIEW

File: account_due_list/views/payment_view.xml [hunk 1]
```diff
@@ -9,25 +9,54 @@
...
+                <field name="invoice_date" readonly="1" optional="show" />
+                <field
+                    name="company_id"
+                    readonly="1"
+                    groups="base.group_multi_company"
+                    optional="hide"
+                />
+                <field name="journal_id" readonly="1" optional="hide" />
+                <field name="move_id" string="Invoice" readonly="1" optional="show" />
+                <field name="invoice_origin" readonly="1" optional="show" />
```
Comment (pedrobaeza) [inline account_due_list/views/payment_view.xml:29]: Why this one is shown?

File: account_due_list/views/payment_view.xml [hunk 2]
```diff
@@ -9,25 +9,54 @@
...
+                <field name="name" readonly="1" optional="show" />
+                <field name="partner_id" readonly="1" optional="show" />
+                <field name="partner_ref" readonly="1" optional="show" />
```
Comment (pedrobaeza) [inline account_due_list/views/payment_view.xml]: And this one? I think it should hide by default

File: account_due_list/views/payment_view.xml [hunk 3]
```diff
@@ -9,25 +9,54 @@
...
+                    optional="hide"
+                />
+                <field
+                    name="analytic_tag_ids"
+                    readonly="1"
+                    widget="many2many_tags"
+                    groups="analytic.group_analytic_tags"
+                    optional="hide"
+                />
                 <field name="debit" readonly="1" sum="Total Debit" />
```
Comment (pedrobaeza) [inline account_due_list/views/payment_view.xml]: Not sure about these 2 being always shown. This is due list, not journal item view, so for me it's more important `balance`, `amount_currency` and `amount_residual` than these 2.

**Review by pedrobaeza — COMMENTED** [review]
Thanks for this rework. It was pretty needed.

**Comment by carlosdauden** [comment]
I agree with your comments, but I wanted to keep the original behavior so as not to disturb anyone.
Changed all except the invoice_origin field which can generate more controversy with sales orders payments flow.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
I think this should filter unposted entries directly, as they are not effects.

**Comment by pedrobaeza** [comment]
@JordiBForgeFlow what do you think about this improvement?

**Comment by carlosdauden** [comment]
The problem is that a journal item can be canceled or in draft and it can be reconciled.
If a journal item is canceled with the intention of modifying something and validating it again and at that moment someone is consulting effects, they may lose complete vision if they do not have the possibility to see these unposted entries.

**Comment by pedrobaeza** [comment]
Then what about presenting only them apart from posted ones?

**Comment by carlosdauden** [comment]
I don't understand:
https://github.com/OCA/account-payment/pull/388/files#diff-51659c468c848f543e7a5321069e828eeaaa88d80b5fb9b81ac22d066c1961feR77-R87

**Comment by carlosdauden** [comment]
And only posted are showed by default:
https://github.com/OCA/account-payment/pull/388/files#diff-51659c468c848f543e7a5321069e828eeaaa88d80b5fb9b81ac22d066c1961feR229

**Comment by pedrobaeza** [comment]
OK, good for me

**Review by cubells — APPROVED** [review]
:+1: 

LGTM

Code review and test

**Pull request merged**